packages/twenty-server/src/modules/code-introspection/code-introspection.service.ts

```diff
@@ -1,105 +1,161 @@
 import { generateFakeValue } from 'src/engine/utils/generate-fake-value';
+import { isDefined } from 'src/utils/is-defined';
 import {
-  CodeIntrospectionException,
-  CodeIntrospectionExceptionCode,
-} from 'src/modules/code-introspection/code-introspection.exception';
+  InputSchema,
+  InputSchemaProperty,
+} from 'src/modules/code-introspection/types/input-schema.type';
 
 @Injectable()
 export class CodeIntrospectionService {
-  private project: Project;
-
-  constructor() {
-    this.project = new Project();
-  }
+  public generateInputData(inputSchema: InputSchema) {
+    return Object.entries(inputSchema).reduce((acc, [key, value]) => {
+      if (isDefined(value.enum)) {
+        acc[key] = value.enum?.[0];
+      } else if (['string', 'number', 'boolean'].includes(value.type)) {
+        acc[key] = generateFakeValue(value.type);
+      } else if (value.type === 'object') {
+        acc[key] = isDefined(value.properties)
+          ? this.generateInputData(value.properties)
+          : {};
+      } else if (value.type === 'array' && isDefined(value.items)) {
+        acc[key] = [generateFakeValue(value.items.type)];
+      }
 
-  public generateInputData(fileContent: string, fileName = 'temp.ts') {
-    const parameters = this.getFunctionInputSchema(fileContent, fileName);
-
-    return this.generateFakeDataFromParams(parameters);
+      return acc;
+    }, {});
   }
 
-  public getFunctionInputSchema(
-    fileContent: string,
-    fileName = 'temp.ts',
-  ): FunctionParameter[] {
-    const sourceFile = this.project.createSourceFile(fileName, fileContent, {
-      overwrite: true,
-    });
-
-    const functionDeclarations = sourceFile.getFunctions();
-
-    if (functionDeclarations.length > 0) {
-      return this.getFunctionParameters(functionDeclarations);
-    }
-
-    const arrowFunctions = sourceFile.getDescendantsOfKind(
-      SyntaxKind.ArrowFunction,
+  public getFunctionInputSchema(fileContent: string): InputSchema {
+    const sourceFile = createSourceFile(
+      'temp.ts',
+      fileContent,
+      ScriptTarget.ESNext,
+      true,
     );
 
-    if (arrowFunctions.length > 0) {
-      return this.getArrowFunctionParameters(arrowFunctions);
-    }
+    const schema: InputSchema = {};
 
-    return [];
+    return this.handleArrowFunctions(schema, sourceFile);
   }
 
-  private getFunctionParameters(
-    functionDeclarations: FunctionDeclaration[],
-  ): FunctionParameter[] {
-    if (functionDeclarations.length > 1) {
-      throw new CodeIntrospectionException(
-        'Only one function is allowed',
-        CodeIntrospectionExceptionCode.ONLY_ONE_FUNCTION_ALLOWED,
-      );
-    }
-
-    const functionDeclaration = functionDeclarations[0];
+  private handleArrowFunctions(schema: InputSchema, sourceFile: SourceFile) {
+    sourceFile.forEachChild((node) => {
+      if (node.kind === SyntaxKind.FunctionDeclaration) {
```

Comment:
so this also handle functions that are not arrow functions? If yes it should be renamed